### PR TITLE
pythonFinder and Pep8 checker: Automatic run on PR, modified jenkins …

### PR DESCRIPTION
**Pull request by brooksturtle**
Jenkins file creates warning if any pep8 errors.

### CODE REVIEW

File: utils/mil_tools/scripts/ReadMe.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+sProgramFinder:
```
Comment (RustyBamboo) [inline utils/mil_tools/scripts/ReadMe.txt:1]: Looks like you got a typo.
Also, you can use markdown to make the readme look nicer.
Convientially the file name should be ```readme.md``` or ```README.md```

File: utils/mil_tools/scripts/pythonFinder.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+pythonList = []
+
+# attempts to open file as text and checks #!/usr/bin/env
+# Just checks the first line
+for program in allFileList:
+    try:
+        for line in open(program):
+            if re.search("#!/usr/bin/env python", line):
+                pythonList.append(program)
+            break
```
Comment (kev-the-dev) [inline utils/mil_tools/scripts/pythonFinder.py:34]: This block should be refactored. You only need to check the FIRST line of each program.

File: utils/mil_tools/scripts/pythonFinder.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+for root, dirs, files in os.walk(directory):
+    for name in files:
+        programName = os.path.join(root, name)
+        allFileList.append(programName)
+
+pythonList = []
+
+# attempts to open file as text and checks #!/usr/bin/env
+# Just checks the first line
+for program in allFileList:
```
Comment (kev-the-dev) [inline utils/mil_tools/scripts/pythonFinder.py:29]: You can also check if the file ends in .py in this loop so you don't need to worry about double entries

File: utils/mil_tools/scripts/pythonFinder.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        for line in open(program):
+            if re.search("#!/usr/bin/env python", line):
+                pythonList.append(program)
+            break
+    except:
+        pass
+
+# checks for ending .py
+for program in allFileList:
+    if '.py' in program and '.pyc' not in program:
```
Comment (kev-the-dev) [inline utils/mil_tools/scripts/pythonFinder.py:40]: Don't need .pyc

**Review by kev-the-dev — CHANGES_REQUESTED** [review]

File: utils/mil_tools/scripts/pythonFinder.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+for program in pythonList:
+    # do not add if duplicate
+    if program in finalList:
+        pass
+    else:
+        finalList.append(program)
+
+counter = 0
+for program in finalList:
```
Comment (kev-the-dev) [inline utils/mil_tools/scripts/pythonFinder.py:54]: This should ideally be just ONE command which runs, checking all files by adding each as an argument.

**Review by kev-the-dev — CHANGES_REQUESTED** [review]

**Comment by kev-the-dev** [comment]
I'm going to have someone else finish this on Monday if it has not been updated.

**Comment by kev-the-dev** [comment]
Checkout #103 to see a simpler and more efficient implementation.

**Pull request closed without merging**